x/poolmanager/types/routes_test.go

```diff
@@ -350,3 +350,66 @@ func TestIntermediateDenoms(t *testing.T) {
 		})
 	}
 }
+
+func TestAddEdge(t *testing.T) {
```

Comment:
Can we add more test cases that tests:
1. start entry already exists, end entry does not
2. start entry and end entry already exists

and test that route is getting appended at the last index properly?

Reply:
Added further test cases here https://github.com/osmosis-labs/osmosis/pull/6895/commits/ebce499d64ae281dca157f27e3f267f7b5051408